pyccel/parser/semantic.py

```diff
@@ -1402,6 +1415,13 @@ def _assign_lhs_variable(self, lhs, d_var, rhs, new_expressions, is_augassign, *
                     new_expressions.append(Allocate(lhs, shape=lhs.alloc_shape, order=lhs.order, status=status))
                 # ...
 
+                # ...
+                # Add memory deallocation for array variables
+                if lhs.is_ndarray:
+                    # Create Deallocate node
+                    self._allocs.append(lhs)
```

Comment:
Is `_allocs` a list? Shouldn't it be a list of lists to handle functions in functions?